### ISSUE TITLE
cmd/pushback: rename to gopherbot?

### ISSUE THREAD

**Issue by bradfitz** [golang/go#14583]
I keep having little idea for Go github automation, but I don't do it because I don't want to run a new server (lazy).

But I realize we already have cmd/pushback running with Github API access, so perhaps we turn cmd/pushback into the general gopherbot server and put more stuff in there?

e.g. I'd like a way to assign issues to people who aren't yet a member of our "go-assignees" Team. Perhaps I could summon @gopherbot to do it for me, like:

"Yo @gopherbot, assign @xxxxxxxxxx"

And gopherbot would manage both the process of inviting them to the team, awaiting the invite, and then doing the github issue assignment.

@adg, thoughts?

**Comment by dgryski** [comment]
@bradfitz GitHub bots can be valuable members of the community.  See for example how Rust handles this: https://www.youtube.com/watch?v=dIageYT0Vgg

**Comment by bradfitz** [comment]
Can you summarize so I don't have to watch a movie?

**Comment by dgryski** [comment]
The current rust bot is called `homu`: valid commands are http://buildbot.rust-lang.org/homu/ ;  Because Rust uses GitHub PRs for all code review, homu provides some of the features that Go gets via Gerrit, OWNERS file,  and the current Trybot infrastructure.  That's from a purely technical perspective.  The talk also covers how having bots handle mundane tasks and also be personified has created a welcoming community in Rust.  That part is trickier to summarize.

**Comment by adg** [comment]
Happy to rename pushback to gopherbot. Although @rsc maintains another service that acts as @gopherbot, so that might be a little confusing.

**Comment by broady** [comment]
Sounds like we should merge the two bots, then?

**Comment by rsc** [comment]
It's fine for both bots to run as gopherbot. Gopherbot also pushes from
Gerrit to Github and that's a separate program too. It's unclear to me that
the programs need to be merged. There is almost no overlap in what they do
and it would be easier to update them individually than as one
mega-program. Given that I think cmd/gopherbot doesn't make much sense.
There are many things that act as gopherbot.

**Comment by adg** [comment]
@bradfitz there's barely anything to cmd/pushback.

We could instead rename the metadata key `pushback-credentials` to `gopherbot-credentials` and provide a package that makes authenticated github requests from managed vms. Then the new bot functionality could leverage the same codebase but run as a separate module.

**Comment by adg** [comment]
Closing this because it seems like these should be separate things, anyway.
